### PR TITLE
Implement killing of subprocesses

### LINKED ISSUE #43: Kill entire ProcessTree
Hi there, 

Thanks for this great lib! I'd like to have an option to kill the entire process tree. I'd do a PR as I need it

### PR DESCRIPTION
**Pull request by M-Patrone**
Fixes #43

### CODE REVIEW

**Comment by M-Patrone** [comment]
Not sure why the test fails, locally it runs (Windows 10)

**Comment by M-Patrone** [comment]
Another thing to note is that killing subprocesses does not work on .net standard 2.0 as there is no API for doing this. The [best API](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=netcore-3.0#System_Diagnostics_Process_Kill_System_Boolean_ ) is the one in .Net Core 3.0 which is finally crossplat

**Comment by Tyrrrz** [comment]
Thanks for the PR @M-Patrone.
I have only glanced through it for now, but I have a suggestion: let's not add an option `killProcessTree`, rather make it implicit. This will avoid some `#if/#endif` directives. `KillProcessAndChildrens` can just no-op if it's not supported.

Regarding CI build, if I remember correctly, tests are targetting `.net45` which is an AppVeyor limitation (it doesn't extract test results from .net core tests), so maybe that's why the tests fail.

**Comment by M-Patrone** [comment]
Tanks for quick response @Tyrrrz. I'm not 100% sure if I really understood you correct, but I removed the property and the optional parameter in the method to kill the entire process tree. If we can, we are going to kill the entire process tree with this pull. Again, in the locally environment, the Test "Execute_Sleep_InSubprocess_CancelLate_Test" runs successfully.

File: CliWrap.Tests/CliTests.cs [hunk 1]
```diff
@@ -131,6 +133,40 @@ public void Execute_EchoSpam_Callback_Test()
...
+
+                Assert.Throws<OperationCanceledException>(() =>
+                {
+                    Cli.Wrap(SleepInSubprocessBat)
+                        .SetCancellationToken(cts.Token)
+                        .Execute();
+                });
+            }
+        }
+
```
Comment (Tyrrrz) [inline CliWrap.Tests/CliTests.cs]: These tests don't seem to be making sure that child processes are killed. In fact, `OperationCanceledException` will be thrown either way. It appears that these tests don't add value on top of the previously existing `Execute_Sleep_CancelEarly_Test` tests.

File: CliWrap.Tests/CliTests.cs [hunk 2]
```diff
@@ -318,6 +354,41 @@ public async Task ExecuteAsync_EchoSpam_Callback_Test()
...
+                Assert.ThrowsAsync<OperationCanceledException>(async () =>
+                {
+                    await Cli.Wrap(SleepBat)
+                        .SetCancellationToken(cts.Token)
+                        .ExecuteAsync();
+                });
+            }
+        }
+
```
Comment (Tyrrrz) [inline CliWrap.Tests/CliTests.cs]: Same story here

**Comment by Tyrrrz** [comment]
>Isn't this quite breaking? Not sure if this is expected by all users

I'd probably expect it to be the original behavior and thus this being a bug fix, but you're right, this might not be the same for other users.

File: CliWrap/Internal/CliProcess.cs [hunk 3]
```diff
@@ -183,5 +192,31 @@ public void Dispose()
             _standardOutputEndSignal.Dispose();
             _standardErrorEndSignal.Dispose();
         }
+#if NET45
+        private static void KillProcessAndChildrens(int pid)
```
[the remark this answers is not in the document]
Comment (M-Patrone) [reply CliWrap/Internal/CliProcess.cs]: As it's recursive, I need the parameter

[the remark this answers is not in the document]
Comment (Tyrrrz) [reply CliWrap/Internal/CliProcess.cs]: Ah, didn't notice, sorry.

**Comment by M-Patrone** [comment]
Thanks @Tyrrrz  for your help and sorry for my late response. I have now implemented the suggested improvements:
- Tests: Check if child processes are terminated
- rename: KillProcessAndChildrens -->KillProcessTree
- removed: TargetFramework .NET Core 3.0

**Comment by Tyrrrz** [comment]
Hi.
Can you please allow committing to your branch (there should be an option in the right section of the PR)? I will commit some small changes and merge.

**Comment by M-Patrone** [comment]
Hi @Tyrrrz 
I think I did it, but I am not sure if thats correct. 
image

**Comment by Tyrrrz** [comment]
Looks like I still can't push, probably because the repository owned by someone else. It's okay, I'll merge and change it locally.

**Comment by M-Patrone** [comment]
Hi @Tyrrrz,  thank you very much. Sorry for that!

**Pull request merged**